src/passes/Inlining.cpp

```diff
@@ -227,10 +238,35 @@ struct FunctionInfoScanner
     info.size = Measurer::measure(curr->body);
 
     if (auto* call = curr->body->dynCast<Call>()) {
+      // If call arguments are function locals read in order, then the code size
+      // always shrinks when the call is inlined. Note that we don't allow
+      // skipping function arguments here, as that can create `drop`
+      // instructions at the call sites, increasing code size.
+      bool shrinks = true;
+      Index nextLocalGetIndex = 0;
+      for (auto* operand : call->operands) {
+        if (auto* localGet = operand->dynCast<LocalGet>()) {
+          if (localGet->index == nextLocalGetIndex) {
+            nextLocalGetIndex += 1;
+          } else {
+            shrinks = false;
+            break;
+          }
+        } else {
+          shrinks = false;
```

Comment:
This is where we could see if the operand is of size 1, and allow it if so (as the old code did).

Reply:
That code still runs as before: when `shrinks == true` we have an early return, but otherwise we still run the old code that checks whether the call size is one larger than the arguments, and sets the flag as `MayNotShrink`. That flag is then used with `-O3`. I added a test to check this.